### PR TITLE
Data models

### LINKED ISSUE #3: Create inventory data models
Take reference from https://github.com/ajc113/inventory/blob/existing/db/schema.rb

Need Location, Flavor, and LocationFlavor model. Location and Flavor are associated through LocationFlavor via a `has_many through` relationship.

Add `archived` boolean flag in Flavor model.

### PR DESCRIPTION
**Pull request by gullariz12**

### CODE REVIEW

File: app/models/flavor.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    yes: 'yes'
+  }.freeze
+
+  enum :instock, STOCK_STATUSES, default: STOCK_STATUSES[:yes]
+
+  validates :name, presence: true, uniqueness: true
+  validates :archived, inclusion: { in: [true, false] }
+  validates :quantity, :inventory, numericality: { only_integer: true }
+
+  has_many :location_flavors
```
Comment (tannakartikey) [inline app/models/flavor.rb]:
```suggestion
  has_many :location_flavors, dependent: :destroy
```

File: app/models/location.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class Location < ApplicationRecord
+  validates :name, presence: true
+
+  has_many :location_flavors
```
Comment (tannakartikey) [inline app/models/location.rb]:
```suggestion
  has_many :location_flavors, dependent: :destroy
```

**Review by tannakartikey — CHANGES_REQUESTED** [review]

File: app/models/flavor.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class Flavor < ApplicationRecord
+  STOCK_STATUSES = {
+    no: 'no',
+    yes: 'yes'
+  }.freeze
+
+  enum :instock, STOCK_STATUSES, default: STOCK_STATUSES[:yes]
```
Comment (tannakartikey) [inline app/models/flavor.rb]: @gullariz12 what do you think about this? Although I don't think we will use this. Most probably we will have `instock` method which will use `quantity` to decide if it's in stock or not.
```suggestion
  enum :instock, [:yes, :no], default: :yes
```

Comment (gullariz12) [reply app/models/flavor.rb]: Agreed with your thoughts. I will remove it from the DB as well we will calculate it on runtime.
Actually, I reference it from the previous scheme and think that we will add a field for it.

File: app/models/flavor.rb
Comment (tannakartikey) [inline app/models/flavor.rb:1]: Let's declare a default scope to return only non archived flavors.

**Review by tannakartikey — CHANGES_REQUESTED** [review]

**Review by tannakartikey — APPROVED** [review]

**Pull request merged**